File: The Long Road/HTML chit.py
import sqlite3
import re


# ******************************************************************************
print ("""<!DOCTYPE html>
<html>
\t<head>
\t\t<meta charset="UTF-8">
\t\t<title>The Long Road</title>
\t</head>
\t<body>
\t\t<table>"""
)

CONN = sqlite3.connect("db/TLR.db")

stmt = "SELECT * FROM v_chit "
stmt += "ORDER BY descrip"
cursor = CONN.cursor()
cursor.execute(stmt)
rows = cursor.fetchall()
for row in rows:
    html_row = "<tr>"
    html_row += f"<td>{row[0]}</td>"
    html_row += f"<td><img src=\"../TLR/{row[3]}\"</td>"
    html_row += f"<td><img src=\"../TLR/{row[4]}\"</td>"
    html_row += f"<td>{row[1]}</td>"
    html_row += f"<td>{row[2]}</td>"
    html_row += "</tr>"
    print(html_row)

CONN.close()

print("""\t\t</table>
\t</body>
</html>
"""
)

File: Zig/The Long Road/html_CHIT.zig
// to compile:
// zig build-exe <foo>.zig -lc -lsqlite3
const std = @import("std");
const print = std.debug.print;

const c = @cImport({
    @cInclude("sqlite3.h");
});

pub fn main() !void {
    // ********** 1: open database
    var db: ?*c.sqlite3 = undefined;
    if (c.sqlite3_open("DB/TLR.db", &db) != c.SQLITE_OK) {
        std.debug.print("Can't open database\n", .{});
        return;
    }
    defer _ = c.sqlite3_close(db);

    const html_1 =
        \\<!DOCTYPE html>
        \\<html>
        \\<head>
        \\   <meta charset="UTF-8">
        \\   <title>{s}</title>
        \\</head>
        \\<body>
        \\<table>
        \\
    ;
    const html_2 =
        \\</table>
        \\</body>
        \\</html>
        \\
    ;

    print(html_1, .{"Images"});
    show_chits(db);
    print(html_2, .{});
}

fn show_chits(db: ?*c.sqlite3) void {
    // ********** 2: prepare statement
    const query =
        \\SELECT * FROM v_chit ORDER BY descrip
    ;

    var stmt: ?*c.sqlite3_stmt = undefined;
    if (c.sqlite3_prepare_v2(db, query, -1, &stmt, null) != c.SQLITE_OK) {
        std.debug.print("SQL error: {s}\n", .{c.sqlite3_errmsg(db)});
        return;
    }
    defer _ = c.sqlite3_finalize(stmt);

    // ********** 3: loop result
    while (c.sqlite3_step(stmt) == c.SQLITE_ROW) {
        const id = c.sqlite3_column_int64(stmt, 0);
        const name = c.sqlite3_column_text(stmt, 1);
        //const faction = c.sqlite3_column_text(stmt, 2);
        const front_img = c.sqlite3_column_text(stmt, 3);
        const back_img = c.sqlite3_column_text(stmt, 4);

        print("   <tr>", .{});
        print("<td>{d}</td><td><img src=\"TLR/{s}\"><img src=\"TLR/{s}\"></td>", .{ id, front_img, back_img });
        print("<td>{s}</td>", .{name});
        print("</tr>\n", .{});
    }
}
